js/decoder.h

```diff
@@ -0,0 +1,435 @@
+        pid->Set(Nan::New("node").ToLocalChecked(), unpack());
+        pid->Set(Nan::New("id").ToLocalChecked(), Integer::New(isolate, read32()));
+        pid->Set(Nan::New("serial").ToLocalChecked(), Integer::New(isolate, read32()));
+        pid->Set(Nan::New("creation").ToLocalChecked(), Integer::New(isolate, read8()));
+        return pid;
+    }
+
+    Local<Value> decodeExport() {
+        auto exp = Object::New(isolate);
+        exp->Set(Nan::New("mod").ToLocalChecked(), unpack());
+        exp->Set(Nan::New("fun").ToLocalChecked(), unpack());
+        exp->Set(Nan::New("arity").ToLocalChecked(), unpack());
+        return exp;
+    }
+
+    Local<Value> unpack() {
+        if (isInvalid) {
+            return Nan::Undefined();
+        }
+
+        if(offset >= size) {
+            THROW("Unpacking beyond the end of the buffer");
+            return Nan::Undefined();
+        }
+
+        const auto type = read8();
+        switch(type) {
+            case SMALL_INTEGER_EXT:
+                return decodeSmallInteger();
+            case INTEGER_EXT:
+                return decodeInteger();
+            case FLOAT_EXT:
+                return decodeFloat();
+            case NEW_FLOAT_EXT:
+                return decodeNewFloat();
+            case ATOM_EXT:
+                return decodeAtom();
+            case SMALL_ATOM_EXT:
+                return decodeSmallAtom();
+            case SMALL_TUPLE_EXT:
+                return decodeSmallTuple();
+            case LARGE_TUPLE_EXT:
+                return decodeLargeTuple();
+            case NIL_EXT:
+                return decodeNil();
+            case STRING_EXT:
+                return decodeString();
+            case LIST_EXT:
+                return decodeList();
+            case MAP_EXT:
+                return decodeMap();
+            case BINARY_EXT:
+                return decodeBinary();
+            case SMALL_BIG_EXT:
+                return decodeSmallBig();
+            case LARGE_BIG_EXT:
+                return decodeLargeBig();
+            case REFERENCE_EXT:
+                return decodeReference();
+            case NEW_REFERENCE_EXT:
+                return decodeNewReference();
+            case PORT_EXT:
+                return decodePort();
+            case PID_EXT:
+                return decodePID();
+            case EXPORT_EXT:
+                return decodeExport();
+            case COMPRESSED:
+                return decodeCompressed();
+            default:
+                THROW("Unsupported erlang term type identifier found");
+                return Nan::Undefined();
+        }
+
+        return Nan::Undefined();
+    }
+private:
+    Isolate* isolate;
+    const uint8_t* const data;
+    const size_t size;
+    bool isInvalid;
+    size_t offset;
+
+    void THROW(const char* msg) {
+        Nan::ThrowError(msg);
+        isInvalid = true;
+        printf("[Error %s:%d] %s\n", __FILE__, __LINE__, msg);
```

Comment:
THROW should probably be a macro, then these `__FILE__` and `__LINE__` would do what you expect.